### PR TITLE
Adding Missing Method Declarations Resolved By __call()

### PR DESCRIPTION
**Pull request by bryceray1121**
Adding phpdoc method declarations to help IDEs autocomplete several magic methods.

### CODE REVIEW

File: src/PhpSpec/ObjectBehavior.php [hunk 1]
```diff
@@ -32,6 +32,9 @@
  * @method void beConstructedThrough($factoryMethod, array $constructorArguments)
  * @method void beAnInstanceOf($class)
  * @method void shouldHaveType($type)
+ * @method void shouldImplement($class)
+ * @method void shouldBeAnInstanceOf($class)
+ * @method void shouldReturnAnInstanceOf($class)
```
Comment (ciaranmcnulty) [inline src/PhpSpec/ObjectBehavior.php]: Not sure this makes sense applied to `$this`?

Comment (bryceray1121) [inline src/PhpSpec/ObjectBehavior.php]: The methods are fulfilled through a magic method.  Isn't that the use-case for adding @method calls to the class definition?

Comment (ciaranmcnulty) [inline src/PhpSpec/ObjectBehavior.php]: I meant for this specific one I can't see `$this->shouldReturnAnInstanceOf` making sense.

The other three are all good additions

Comment (bryceray1121) [inline src/PhpSpec/ObjectBehavior.php]: Oh ok, I just copied the method names from the "Type Matcher" section of the documentation: http://www.phpspec.net/en/latest/cookbook/matchers.html

Comment (ciaranmcnulty) [inline src/PhpSpec/ObjectBehavior.php]: The method exists but in the case of calling it on the object being tested, I don't think it's something to encourage.

If you remove that line I'll merge the others.

Comment (ciaranmcnulty) [inline src/PhpSpec/ObjectBehavior.php]: Can you please remove `shouldReturnAnInstanceOf`?

**Comment by ciaranmcnulty** [comment]
Can you address my last comment above and resolve the merge conflict please?

**Comment by ciaranmcnulty** [comment]
I'm closing this due to inactivity, please feel free to reopen if you have time to address the comments

**Pull request closed without merging**